api/plan.go

```diff
@@ -120,6 +122,52 @@ func (e FeatureType) String() string {
 	return ""
 }
 
+// Name returns a readable name of the feature
+func (e FeatureType) Name() string {
+	switch e {
+	case FeatureBackwardCompatibilty:
+		return "Backward compatibility"
+	case FeatureSchemaDrift:
+		return "Schema drift"
+	case FeatureTaskScheduleTime:
+		return "Task schedule time"
+	case FeatureMultiTenancy:
+		return "Multi-tenancy"
+	case FeatureDBAWorkflow:
+		return "DBA workflow"
+	case FeatureDataSource:
+		return "Data source"
+	case FeatureApprovalPolicy:
+		return "Approval policy"
+	case FeatureBackupPolicy:
+		return "Backup policy"
+	case FeatureRBAC:
+		return "RBAC"
+	case Feature3rdPartyLogin:
+		return "3rd party login"
+	}
+	return ""
+}
+
+// AccessErrorMessage returns a error message with feature name and minimum supported plan.
+func (e FeatureType) AccessErrorMessage() string {
+	plan := e.minimumSupportedPlan()
+	return fmt.Sprintf("%s is a %s feature, please upgrade to access it.", e.Name(), plan.String())
+}
+
+// minimumSupportedPlan will find the minimum plan which support the target feature.
+func (e FeatureType) minimumSupportedPlan() PlanType {
+	index := 0
+	for i, enabled := range FeatureMatrix[e] {
+		if enabled {
+			index = i
```

Comment:
nit: return directly

if enabled {
  return PlanType[i]
}

But how do we handle "bb.feature.data-source" when all columns are false?

Reply:
We're not handling it right now, so I prefer to return the highest plan if not found. Updated now.